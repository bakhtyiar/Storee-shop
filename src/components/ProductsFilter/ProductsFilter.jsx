import React, {useEffect, useState} from 'react';
import {Button, Form} from "react-bootstrap";
import styled from "styled-components";
import {getCategories} from "../../utils/server-api/products/products";
import {replaceDashToSpace} from "../../utils/str/str";

const StyledSection = styled.section`
  display: flex;
  flex-wrap: wrap;
  gap: 8px;
  margin: 8px 0;
`;

function ProductsFilter({searchParams, setSearchParams}) {
    const [isLoading, setIsLoading] = useState(true);
    const [categories, setCategories] = useState([]);

    useEffect(() => {
        setIsLoading(true);
        getCategories()
            .then(data => setCategories(data))
            .finally(() => setIsLoading(false));
    }, []);

    const handleClick = (clickedCategory) => {
        if (searchParams.get('category') === clickedCategory) {
            searchParams.delete('category');
            setSearchParams({...searchParams});
        } else
            setSearchParams({...searchParams, 'category': clickedCategory})
    }

    return (
        <Form className='mb-4' data-testid={"products-filter"}>
            <h3>Filter</h3>
            <h4>Categories</h4>
            <StyledSection>
                {!isLoading &&
                    categories.map((item, index) => (<Button
                        variant={item === searchParams.get('category') ? "primary" : "outline-primary"}
                        id={`category-btn-${item}`}
                        key={`category-btn-${index}`}
                        onClick={() => handleClick(item)}
                    >
                        {replaceDashToSpace(item.toUpperCase())}
                    </Button>))
                }
            </StyledSection>
            {/*<Button variant="primary" type="submit">*/}
            {/*    Filter*/}
            {/*</Button>*/}
        </Form>
    );
}

export default ProductsFilter;