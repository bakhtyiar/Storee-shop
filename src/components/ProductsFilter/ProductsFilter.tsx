import React, {useEffect, useState} from 'react';
import {Button, Form} from "react-bootstrap";
import styled from "styled-components";
import {getCategories} from "../../utils/server-api/products/products";
import {replaceDashToSpace} from "../../utils/str/str";
import {ICategories} from "../../utils/server-api/products/products.types";

const StyledSection = styled.section`
  display: flex;
  flex-wrap: wrap;
  gap: 8px;
  margin: 8px 0;
`;

function ProductsFilter({
    searchParams,
    setSearchParams
}: any) {
    const [isLoading, setIsLoading] = useState(true);
    const [categories, setCategories] = useState<ICategories>([]);

    useEffect(() => {
        setIsLoading(true);
        getCategories()
            .then(data => setCategories(data))
            .finally(() => setIsLoading(false));
    }, []);

    const handleClick = (clickedCategory: any) => {
        if (searchParams.get('category') === clickedCategory) {
            searchParams.delete('category');
            setSearchParams({...searchParams});
        } else
            setSearchParams({...searchParams, 'category': clickedCategory})
    }

    return (
        <Form className='mb-4' data-testid={"products-filter"}>
            
            <h3>Filter</h3>
            
            <h4>Categories</h4>
            
            <StyledSection data-testid="categories-filter">
                {!isLoading &&
                    categories.map((item, index) => (<Button
                        variant={item === searchParams.get('category') ? "primary" : "outline-primary"}
                        id={`category-btn-${item}`}
                        key={`category-btn-${index}`}
                        onClick={() => handleClick(item)}
                        data-testid={`category-${item}`}
                    >
                        {replaceDashToSpace(item.toUpperCase())}
                    </Button>))
                }
            </StyledSection>
            {/*<Button variant="primary" type="submit">*/}
            {/*    Filter*/}
            {/*</Button>*/}
        </Form>
    );
}

export default ProductsFilter;