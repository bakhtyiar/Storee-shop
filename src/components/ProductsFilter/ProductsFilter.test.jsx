import {screen} from "@testing-library/react";
import {renderWithRouter} from "../../tests/helpers/renderWithRouter";
import ProductsFilter from "./ProductsFilter";
//todo: write interactive tests
describe("Basic tests", () => {
    test("Match snapshot", () => {
        renderWithRouter(<ProductsFilter/>);
        expect(screen.getByTestId("products-filter")).toMatchSnapshot();
    })
})