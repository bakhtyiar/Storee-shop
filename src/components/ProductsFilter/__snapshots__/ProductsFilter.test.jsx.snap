// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic tests Match snapshot 1`] = `
<form
  class="mb-4"
  data-testid="products-filter"
>
  <h3>
    Filter
  </h3>
  <h4>
    ICategories
  </h4>
  <section
    class="sc-gswNZR dKDtuy"
  />
</form>
`;
